packages/blocks/src/checkbox-input/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { RawHTML, useEffect } from '@wordpress/element';
+import classnames from 'classnames';
+import { isEmpty } from 'lodash';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useField } from '@crowdsignal/form';
+import { ErrorMessage, CheckboxAnswer, FormInputWrapper } from '../components';
+import { useColorStyles, useBorderStyles } from '@crowdsignal/styles';
+
+const CheckboxInput = ( { attributes, className } ) => {
+	const { error, fieldValue, onChange } = useField( {
+		fieldName: `q_${ attributes.clientId }[text]`,
+		validation: ( value ) => {
+			if ( attributes.mandatory && ! value ) {
+				return __( 'This field is required', 'blocks' );
+			}
+		},
+	} );
+
+	const isSelected = fieldValue === attributes.checkedText;
+	const value = isSelected
+		? attributes.checkedText
+		: attributes.uncheckedText;
+	const classes = classnames(
+		className,
+		'crowdsignal-forms-checkbox-input',
+		'is-inline',
+		{
+			'is-required': attributes.mandatory,
+			'is-error': error,
+		}
+	);
+
+	useEffect( () => {
+		if ( isEmpty( fieldValue ) ) {
+			onChange( attributes.uncheckedText );
+		}
+	}, [] );
+
+	const onChangeHandler = () => {
+		if ( isSelected ) {
+			onChange( attributes.uncheckedText );
+		} else {
+			onChange( attributes.checkedText );
+		}
+	};
+
+	return (
+		<FormInputWrapper
+			style={ {
+				...useColorStyles( attributes ),
+				...useBorderStyles( attributes ),
+			} }
+			className={ classes }
+		>
+			<CheckboxAnswer
```

Comment:
After considering it for a bit, I'm not sure if `CheckboxAnswer` really makes sense here - it's only a lightweight wrapper that was meant for multiple choice question's block choices.  

And since it doesn't even support `isRequired`, let's just use the regular `FormCheckbox` and `FormCheckbox.Label` components and put everything together here.  
I don't think you want to have additional `<label>` components either, as clicking on them would no longer focus/trigger the correct input.

Reply:
Made this change. Good suggestion.